tests/gem5/regression_tests/configs/matrix-multiply.py

```diff
@@ -0,0 +1,138 @@
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+This gem5 configuation script creates a simple board to run a matrix multiply
+workload. The ISA of the board and workload depends on the argument passed
+into the config.
+
+This setup is close to the simplest setup possible using the gem5 library. It
+does not contain any kind of caching, IO, or any non-essential components.
+
+Usage
+-----
+
+```
+cd tests
+./main.py run gem5/regression_tests -t <number of threads to run tests with>
+
+```
+"""
+
+import argparse
+
+from gem5.components.boards.simple_board import SimpleBoard
+from gem5.components.cachehierarchies.classic.no_cache import NoCache
+from gem5.components.memory.simple import SingleChannelSimpleMemory
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.isas import get_isa_from_str
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "--isa",
+    required=True,
+    choices=["arm", "x86", "riscv"],
+    help="Enter the isa to use.",
+)
+
+args = parser.parse_args()
+
+# This check ensures the gem5 binary contains the ISA target. If not,
+# an exception will be thrown.
+requires(isa_required=get_isa_from_str(args.isa))
+
+# In this setup we don't have a cache. `NoCache` can be used for such setups.
+cache_hierarchy = NoCache()
+
+# We use a simple memory system. Taken from
+# tests/gem5/replacement_policies/configs/run_replacement_policy.py
+memory = SingleChannelSimpleMemory(
+    latency="30ns",
+    latency_var="0ns",
+    bandwidth="12.8GiB/s",
+    size="512MiB",
+)
+
+# We use a simple Timing processor with one core.
+processor = SimpleProcessor(
+    cpu_type=CPUTypes.TIMING, isa=get_isa_from_str(args.isa), num_cores=1
+)
+
+# The gem5 library simple board which can be used to run simple SE-mode
+# simulations.
+board = SimpleBoard(
+    clk_freq="3GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+# Here we set the workload. In this case we want to run a matrix multiply
+# workload.
+board.set_workload(
+    obtain_resource(
+        f"{args.isa}-matrix-multiply-run", resource_version="1.0.0"
+    )
+)
+
+# Lastly we run the simulation.
+simulator = Simulator(board=board)
+simulator.run()
+
+stats = simulator.get_stats()
+
+# Compare values in the current run against reference IPC values calculated
+# using simInsts / numCycles. The reference IPC values were obtained by running
+# the tests on gem5 v25.0.0.0.
+
+reference_ipcs = {
+    "arm": 0.006344991461591237,
+    "riscv": 0.006273970221306115,
+    "x86": 0.005040373616806491,
```

Comment:
These seem too low.

Reply:
It doesn't really matter 25.0.0.0 is fine. Just document what the commit hash was in the file, please